From: Bjorn Helgaas <bhelgaas@google.com>
Date: Fri, 9 Nov 2012 20:35:01 -0700
Subject: PCI: Remove useless "!dev" tests
Patch-mainline: v3.8-rc1
Git-commit: 1452cd76a97bf7b93a015586dcabc73fd935e692
References: bnc#908322 FATE#317535

No need to check "!dev" when the caller should always supply a valid
pointer.  If the caller *doesn't* supply a valid pointer, it probably
won't check for a failure return either.  This way we'll oops and get a
backtrace.

Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/pci/iov.c |   12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

--- a/drivers/pci/iov.c
+++ b/drivers/pci/iov.c
@@ -716,10 +716,9 @@ EXPORT_SYMBOL_GPL(pci_sriov_migration);
  */
 int pci_num_vf(struct pci_dev *dev)
 {
-	if (!dev || !dev->is_physfn)
+	if (!dev->is_physfn)
 		return 0;
-	else
-		return dev->sriov->nr_virtfn;
+	return dev->sriov->nr_virtfn;
 }
 EXPORT_SYMBOL_GPL(pci_num_vf);
 
@@ -737,7 +736,7 @@ EXPORT_SYMBOL_GPL(pci_num_vf);
  */
 int pci_sriov_set_totalvfs(struct pci_dev *dev, u16 numvfs)
 {
-	if (!dev || !dev->is_physfn || (numvfs > dev->sriov->total))
+	if (!dev->is_physfn || (numvfs > dev->sriov->total))
 		return -EINVAL;
 
 	/* Shouldn't change if VFs already enabled */
@@ -760,13 +759,12 @@ EXPORT_SYMBOL_GPL(pci_sriov_set_totalvfs
  */
 int pci_sriov_get_totalvfs(struct pci_dev *dev)
 {
-	if (!dev || !dev->is_physfn)
+	if (!dev->is_physfn)
 		return -EINVAL;
 
 	if (dev->sriov->drvttl)
 		return dev->sriov->drvttl;
-	else
-		return dev->sriov->total;
+	return dev->sriov->total;
 }
 EXPORT_SYMBOL_GPL(pci_sriov_get_totalvfs);
 
